logos/logos-workernode/tests/test_model_cache.py

```
@@ -199,3 +234,100 @@ async def test_scan_existing_on_init(ram_cache_env):
+    assert result == source_parent
+    assert manifest_only_source["model_name"] not in cache.cached_models()
+
+
+def test_scan_existing_evicts_incomplete_cache(tmp_path):
+    """A tmpfs entry with a 0-byte refs/<branch> (ENOSPC mid-download) is evicted.
+
+    Without this, a worker that crashed once would loop forever: scan
+    re-claims the broken directory as 'cached', ensure_cached returns the
+    tmpfs HF_HOME, vLLM tries to use it, fails again on the same ref write.
+    """
+    import os as _os
```

Flake8 (7.3.0)

[error] 'os as _os' imported but unused

(F401)